src/main/resources/static/javascript/mainApp/grouping.controller.js

```diff
@@ -418,17 +409,21 @@
             $scope.membersInList = "";
             $scope.multiRemoveResults = [];
             $scope.waitingForImportResponse = false;
-            resetCheckboxes();
         };
 
         $scope.resetFields = () => {
             $scope.manageMembers = "";
-            $scope.membersInCheckboxList = {};
             $scope.paginationPageChange = true;
             $scope.checkedBoxes = 0;
-            $scope.pageSelected = false;
             $scope.waitingForImportResponse = false;
             $scope.errorDismissed = true;
+            $scope.pageSelected = false;
+            for (const mem in $scope.membersInCheckboxList) {
+                //codacy will complain without if statement
+                if (typeof $scope.membersInCheckboxList[String(mem)] !== "undefined") {
+                    $scope.membersInCheckboxList[String(mem)] = false;
+                }
+            }
```

Comment:
Use a .map function here instead to get around the Codacy workaround.

Reply:
Using this map function shows as an undefined function.